components/layout_2020/display_list/stacking_context.rs

```diff
@@ -272,7 +290,10 @@ impl StackingContext {
         while child_fragments.peek().map_or(false, |child| {
             child.section == StackingContextSection::BlockBackgroundsAndBorders
         }) {
-            child_fragments.next().unwrap().build_display_list(builder);
+            child_fragments
+                .next()
+                .unwrap()
+                .build_display_list(builder, root_background_style_override.take());
```

Comment:
Can this cause the canvas background to be painted above negative z-index stacking contexts? Per https://drafts.csswg.org/css2/zindex.html the canvas background should always be bottom-most.

Reply:
Assuming I understand your question correct, this commit shouldn't modify the that backgrounds are painted. It only changes the size of the `painting_area` and the style of the painted background.

Reply:
No, this commit should also change the painting order. For example:

```html
<body style="border: solid; background: green">
<p style="position: absolute; z-index: -1">Text
```

My understanding of the spec is that the painting order should be: green canvas background, then text inside a stacking context with negative z-index, then border of an block-level element (that does not generate a stacking context).

But my reading of this PR is that it makes the order: text, then background (incorrectly masking text), then border. Am I reading correctly?

Reply:
Ah, I see what you mean! I thought you were asking if this commit already modified the painting order. It doesn't change the current painting order, but it is true that the root background is currently not always placed into the proper section of the stacking context. This commit could also fix that problem though.